src/mono/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs

```diff
@@ -198,10 +201,14 @@ public bool Join(int millisecondsTimeout)
             return JoinInternal(this, millisecondsTimeout);
         }
 
+        [MemberNotNull(nameof(_waitInfo))]
         private void Initialize()
         {
             InitInternal(this);
 
+#if TARGET_UNIX
+            _waitInfo = new WaitSubsystem.ThreadWaitInfo(this);
+#endif
             // TODO: This can go away once the mono/mono mirror is disabled
```

Comment:
I think you will need to fix this TODO. `_startHelper` can be null here given the change in the default constructor.

Reply:
Ah, yep. The mirror, thankfully, is now disabled :)